packages/jsapi-nodejs/src/loaderUtils.ts

```diff
@@ -27,73 +27,54 @@ export type LoadModuleOptions = {
  * @param serverPaths The paths of the modules on the server.
  * @param download Whether to download the modules from the server. If set to false,
  * it's assumed that the modules have already been downloaded and still exist in
- * the storage directory.
+ * the storage directory. If set to `true` or a `PostDownloadTransform` function,
+ * the modules will be downloaded and stored. If set to a `PostDownloadTransform`
+ * function, the downloaded content will be passed to the function and the result
+ * saved to disk.
  * @param storageDir The directory to store the downloaded modules.
- * @param sourceModuleType module format from the server.
- * @param targetModuleType (optional) module format to be exported. Defaults to
- * sourceModuleType.
- * @param esbuildOptions (optional) Additional options to pass to esbuild.
+ * @param targetModuleType The type of module to load. Can be either 'esm' or 'cjs'.
  * @returns The default export of the first module in `serverPaths`.
  */
 export async function loadModules<TMainModule>({
   serverUrl,
   serverPaths,
   download,
   storageDir,
-  sourceModuleType,
-  targetModuleType = sourceModuleType,
-  esbuildOptions,
+  targetModuleType,
 }: LoadModuleOptions): Promise<TMainModule> {
   polyfillWs();
 
   const serverStorageDir = path.join(storageDir, urlToDirectoryName(serverUrl));
-  const targetDir = path.join(serverStorageDir, 'target');
-
-  if (download) {
-    const needsTranspile = sourceModuleType !== targetModuleType;
-    const sourceDir = path.join(serverStorageDir, 'source');
 
-    ensureDirectoriesExist(
-      needsTranspile ? [sourceDir, targetDir] : [targetDir]
-    );
+  if (download !== false) {
+    ensureDirectoriesExist([serverStorageDir]);
 
     // Download from server
     const serverUrls = serverPaths.map(
       serverPath => new URL(serverPath, serverUrl)
     );
-    const contents = await Promise.all(
+    let contents = await Promise.all(
       serverUrls.map(url => downloadFromURL(url))
     );
 
+    // Post-download transform
+    if (typeof download === 'function') {
+      contents = contents.map((content, i) =>
+        download(serverPaths[i], content)
+      );
+    }
+
     // Write to disk
-    const downloadPaths = getDownloadPaths(
-      needsTranspile ? sourceDir : targetDir,
-      serverPaths
-    );
+    const downloadPaths = getDownloadPaths(serverStorageDir, serverPaths);
```

Comment:
Should mention this change in the PR description as well - it's now downloading to the `serverStorageDir` instead of a `target` subdirectory.